### ISSUE TITLE
Fasta crash: Unhandled scope

### ISSUE THREAD

**Issue by lukechurch** [dart-lang/sdk#29978]
@peter-ahe-google
ID: 3058b87002be3a5f67c6d991d047354365fd1494
```
Unhandled exception:
Unhandled: Scope
#0      internalError (package:front_end/src/fasta/errors.dart:38:5)
#1      BodyBuilder.toValue (package:front_end/src/fasta/kernel/body_builder.dart:226:14)
#2      BodyBuilder.endArguments (package:front_end/src/fasta/kernel/body_builder.dart:682:24)
#3      Parser.parseArguments (package:front_end/src/fasta/parser/parser.dart:3446:14)
#4      Parser.parseArgumentsOpt (package:front_end/src/fasta/parser/parser.dart:3415:14)
#5      Parser.parseSend (package:front_end/src/fasta/parser/parser.dart:3395:13)
#6      Parser.parseSendOrFunctionLiteral (package:front_end/src/fasta/parser/parser.dart:3219:14)
#7      Parser.parsePrimary (package:front_end/src/fasta/parser/parser.dart:2972:14)
#8      Parser.parsePrecedenceExpression (package:front_end/src/fasta/parser/parser.dart:2836:21)
#9      Parser.parseExpressionWithoutCascade (package:front_end/src/fasta/parser/parser.dart:2793:11)
#10     Parser.parseConditionalExpressionRest (package:front_end/src/fasta/parser/parser.dart:2802:13)
#11     Parser.parsePrecedenceExpression (package:front_end/src/fasta/parser/parser.dart:2854:19)
#12     Parser.parsePrecedenceExpression (package:front_end/src/fasta/parser/parser.dart:2826:19)
#13     Parser.parseExpression (package:front_end/src/fasta/parser/parser.dart:2784:11)
#14     Parser.parseParenthesizedExpression (package:front_end/src/fasta/parser/parser.dart:3057:13)
#15     Parser.parseParenthesizedExpressionOrFunctionLiteral (package:front_end/src/fasta/parser/parser.dart:3045:15)
#16     Parser.parsePrimary (package:front_end/src/fasta/parser/parser.dart:3008:14)
#17     Parser.parseUnaryExpression (package:front_end/src/fasta/parser/parser.dart:2942:14)
#18     Parser.parsePrecedenceExpression (package:front_end/src/fasta/parser/parser.dart:2811:13)
#19     Parser.parsePrecedenceExpression (package:front_end/src/fasta/parser/parser.dart:2860:15)
#20     Parser.parseExpression (package:front_end/src/fasta/parser/parser.dart:2784:11)
#21     Parser.parseReturnStatement (package:front_end/src/fasta/parser/parser.dart:2528:15)
#22     Parser.parseStatementX (package:front_end/src/fasta/parser/parser.dart:2445:14)
#23     Parser.parseStatement (package:front_end/src/fasta/parser/parser.dart:2433:20)
#24     Parser.parseFunctionBody (package:front_end/src/fasta/parser/parser.dart:2363:15)
#25     DietListener.parseFunctionBody (package:front_end/src/fasta/source/diet_listener.dart:545:14)
#26     DietListener.buildFunctionBody (package:front_end/src/fasta/source/diet_listener.dart:433:5)
#27     DietListener.endMethod (package:front_end/src/fasta/source/diet_listener.dart:401:5)
#28     Parser.parseMethod (package:front_end/src/fasta/parser/parser.dart:2154:14)
#29     Parser.parseMember (package:front_end/src/fasta/parser/parser.dart:2046:11)
#30     Parser.parseClassBody (package:front_end/src/fasta/parser/parser.dart:1939:15)
#31     Parser.parseClass (package:front_end/src/fasta/parser/parser.dart:1071:13)
#32     Parser.parseClassOrNamedMixinApplication (package:front_end/src/fasta/parser/parser.dart:1032:14)
#33     Parser._parseTopLevelDeclaration (package:front_end/src/fasta/parser/parser.dart:357:14)
#34     Parser.parseTopLevelDeclaration (package:front_end/src/fasta/parser/parser.dart:344:13)
#35     Parser.parseUnit (package:front_end/src/fasta/parser/parser.dart:336:15)
#36     SourceLoader.buildBody (package:front_end/src/fasta/source/source_loader.dart:146:14)
<asynchronous suspension>
#37     Loader.buildBodies (package:front_end/src/fasta/loader.dart:117:13)
<asynchronous suspension>
#38     KernelTarget.buildProgram (package:front_end/src/fasta/kernel/kernel_target.dart:282:20)
<asynchronous suspension>
#39     CompileTask.compile (package:front_end/src/fasta/fasta.dart:164:38)
<asynchronous suspension>
#40     compile.<anonymous closure> (package:front_end/src/fasta/fasta.dart:101:25)
<asynchronous suspension>
#41     CompilerContext.withGlobalOptions.<anonymous closure> (package:front_end/src/fasta/compiler_context.dart:45:33)
#42     _rootRun (dart:async/zone.dart:1120)
#43     _CustomZone.run (dart:async/zone.dart:1001)
#44     runZoned (dart:async/zone.dart:1467)
#45     CompilerContext.withGlobalOptions (package:front_end/src/fasta/compiler_context.dart:45:12)
#46     CompilerCommandLine.withGlobalOptions (package:front_end/src/fasta/compiler_command_line.dart:120:28)
#47     compile (package:front_end/src/fasta/fasta.dart:94:38)
<asynchronous suspension>
#48     compileEntryPoint (package:front_end/src/fasta/fasta.dart:53:11)
<asynchronous suspension>
#49     main (file:///home/lukechurch/GitRepos/dart-sdk/sdk/pkg/front_end/tool/_fasta/compile.dart:7:33)
#50     _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:263)
#51     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:151)

```

**Comment by peter-ahe-google** [comment]
Repro:

```
foo(a, b) => null;

main() {
  foo(null, f() {});
}
```

**Comment by peter-ahe-google** [comment]
Proposed fix in [CL 2968093003](https://codereview.chromium.org/2968093003).
